content/en/docs/usage/kraft_advanced.md

```diff
@@ -0,0 +1,732 @@
+                    Mimas 0.7.0~1939bce
+Hello world!
+Arguments:  "/media/stefan/projects/unikraft/my-helloworld-app/build/helloworld_kvm-x86_64" "console=ttyS0"
+```
+
+You can use the same arguments that you use with [`kraft init`](docs/usage/kraft_advanced/#kraft-init), [`kraft configure`](docs/usage/kraft_advanced/#kraft-configure), [`kraft build`](docs/usage/kraft_advanced/#kraft-configure) and [`kraft run`](docs/usage/kraft_advanced/#kraft-configure).
+
+Behind the scenes, the `kraft up` command does the following:
+
+1. It downloads the `helloworld` application repository in the `my-helloworld-app` directory (i.e. runs `kraft init -t helloworld my-helloworld-app`).
+1. It configures the repository, resulting in a `.config` file (i.e. runs `kraft configure`).
+1. It builds the required components, resulting in the `build/helloworld_kvm-x86_64 unikernel image` (i.e. runs `kraft build`).
+1. It runs the image, resulting in `QEMU/KVM` being run, with the `Hello world!` message getting printed (i.e. runs `kraft run`).
+
+All that magic is done using one command.
+
+### Overview of the `kraft up` Command
+
+You can get a full list of features by running `kraft up -h` or `kraft up --help`:
+
+```text
+Usage: kraft up [OPTIONS] NAME
+
+  Configures, builds and runs an application for a selected architecture and
+  platform.
+
+Options:
+  -w, --workdir PATH         Specify an alternative directory for the
+			     application [default is cwd].
+  -t, --template NAME        Use an existing application as a template.
+			     [required]
+  -p, --plat TEXT            Target platform.
+  -m, --arch TEXT            Target architecture.
+  -i, --initrd TEXT          Provide an init ramdisk.
+  -B, --background           Run in background.
+  -P, --paused               Run the application in paused state.
+  -g, --gdb INTEGER          Run a GDB server for the guest on specified port.
+  -d, --dbg                  Use unstriped unikernel
+  -n, --virtio-nic TEXT      Attach a NAT-ed virtio-NIC to the guest.
+  -b, --bridge TEXT          Attach a NAT-ed virtio-NIC an existing bridge.
+  -V, --interface TEXT       Assign host device interface directly as virtio-
+			     NIC to the guest.
+  -D, --dry-run              Perform a dry run.
+  -M, --memory INTEGER       Assign MB memory to the guest.
+  -s, --cpu-sockets INTEGER  Number of guest CPU sockets.
+  -c, --cpu-cores INTEGER    Number of guest cores per socket.
+  -F, --force                Overwrite any existing files in current working
+			     directory.
+  -j, --fast                 Use all CPU cores to build the application.
+  -M, --with-makefile        Create a Unikraft compatible Makefile.
+  -h, --help                 Show this message and exit.
+```
+
+## `kraft init`
+
+You can initialise an empty application, or download an existing one by using `kraft init`.
+You can choose to create a compatible Makefile for the application by using the `-M, --with-makefile` option.
```

Comment:
This doesn't seem to work:

```console
razvan@yggdrasil:~/.../unikraft/test/zuzu$ kraft init -M -t helloworld-cpp my-helloworld-cpp
 100.00% :::::::::::::::::::::::::::::::::::::::: |       19 /       19 |:  app/helloworld-cpp@0.11.0                                                                                 
[WARNING ] Could not find manifest: app/helloworld-cpp
 100.00% :::::::::::::::::::::::::::::::::::::::: |      166 /      166 |:  lib/musl@0.11.0                                                                                           
[INFO    ] Initialized new unikraft application: /home/razvan/projects/unikraft/test/zuzu/my-helloworld-cpp
razvan@yggdrasil:~/.../unikraft/test/zuzu$ cd my-helloworld-cpp/
razvan@yggdrasil:~/.../test/zuzu/my-helloworld-cpp$ ls
helloworld.cpp  kraft.yaml  Makefile.uk  README.md
```

Also, not sure why the manifest file for `app-helloworld-cpp` is not available.

Reply:
Indeed this does not work, I'll just remove the `-M` option.